lib/operator/subtract.js

```diff
@@ -87,12 +83,8 @@ class SubtractBaseObserver { // implements CollectionObserver
   }
 
   removed(items, coll) {
-    for (let item of items) {
-      if (this.resultColl._collSubtract.contains(item)) {
-        return;
-      }
-      this.resultColl.removeEach(item);
-    }
+    var removeItems = items.filter(item => !this.resultColl._collSubtract.contains(item));
+    this.resultColl.removeAll(removeItems);
```

Comment:
That change introduces a bug. If the base collection contains the removed value multiple times, and the substract coll contains contains it only once, now `removeAll()` removes only a single instance, whereas `removeEach()` removed all of them. The old behavior was correct, the new one leaves substracted values in the output.

Please add a test for this.

Ironically, the old implenentation of `added()` has exactly the same bug, and your new implementation of `added()` fixes it. Funny.

Reply:
I do not understand the goal of why you change the implemenation of all these base operations.

Reply:
No, this is for when you remove items from the base collection; you only want to remove them once from the result collection in this case. (Although, they should not be in the base collection in the first place, so the filter is unnecessary.)

Reply:
...they should not be in the *result* collection...

Reply:
Oh, I see. I didn't have enough context in the patch to notice that this is the base coll.

Good point on the subtract coll being irrelevant.

Reply:
```
@@ -87,12 +83,8 @@ class SubtractBaseObserver { // implements CollectionObserver
```
should have been the giveaway.

Would you like me to remove the subtract coll check?